integration/query_test.go

```diff
@@ -503,3 +503,67 @@ func TestQueryNonExistingCollection(t *testing.T) {
 	require.NoError(t, err)
 	require.Len(t, actual, 0)
 }
+
+func TestQueryBatchSize(t *testing.T) {
```

Comment:
Why it wasn't possible to write a compat test for this case?

Reply:
That test and `TestGetMoreErrors` are a bit different in the way how they work. We are expecting errors here and there, so we don't need to insert any data to run tests. The current compatibility tests setup assumes that there would be some data providers and makes it difficult to write proper tests. There are some similar tests (`TestQueryBadMaxTimeMSType`, `TestQueryBadSortType`) that were not moved to compat yet. We can work on that kind of tests in a separate issue.

Reply:
Got it, thanks! Sounds reasonable! Let's add a comment about it? (To clarify that we don't plan to move it to compat tests)

Reply:
Just noting that it might be easier to just write compat for testing if we are expecting errors here and there. We can setup compat the way we need it so empty collection should be easy too :hugs: 

Anyway I think now our test-integration-mongodb runs integration tests too so error itself is also tested against compat like system I think.

Reply:
Moved to compat both tests.